unittest/test_diskann.cpp

```diff
@@ -423,4 +423,62 @@ TEST_P(DiskANNTest, config_test) {
 
     range_search_conf_to_test.beamwidth = 129;
     check_config_error<knowhere::DiskANNQueryByRangeConfig>(range_search_conf_to_test);
+}
+
+TEST_P(DiskANNTest, build_config_test) {
+    std::string test_dir = kDir + "/build_config_test";
+    std::string test_data_path = test_dir + "/test_raw_data";
+    std::string test_index_dir = test_dir + "/test_index";
+    ASSERT_TRUE(fs::create_directory(test_dir));
+
+    // node size > diskann sector 
+    std::vector<std::tuple<uint32_t, uint32_t>> illegal_dim_and_R_set {
+        std::make_tuple(1024, 32),
+        std::make_tuple(400, 640),
+        std::make_tuple(960, 128)
+    };
+    knowhere::Config illegal_cfg;
+    knowhere::DiskANNBuildConfig illegal_build_config;
+
+    for (auto& dim_and_R : illegal_dim_and_R_set) {
+        fs::remove(test_data_path);
+        fs::remove(test_index_dir);
+
+        uint32_t data_dim = std::get<0>(dim_and_R);
+        uint32_t data_n = 1000;
+        std::vector<float> raw_data(data_dim * data_n);
+        WriteRawDataToDisk(test_data_path, raw_data.data(), data_n, data_dim);
+
+        illegal_build_config = build_conf; // init
+        illegal_build_config.data_path = test_data_path;
+        illegal_build_config.max_degree = std::get<1>(dim_and_R);
+        illegal_cfg.clear();
+        knowhere::DiskANNBuildConfig::Set(illegal_cfg, illegal_build_config);
+        knowhere::IndexDiskANN<float> test_diskann(test_index_dir, metric_,
+                                                   std::make_unique<knowhere::LocalFileManager>());
+        EXPECT_THROW(test_diskann.BuildAll(nullptr, illegal_cfg), knowhere::KnowhereException);
+
+        fs::remove(test_data_path);
+        fs::remove(test_index_dir);
+    }
+    // data size != file size
+    {
+        fs::remove(test_data_path);
+        fs::remove(test_index_dir);
+        fs::copy_file(kRawDataPath, test_data_path);
+        std::ofstream writer(test_data_path, std::ios::binary);
+        writer.seekp(0, std::ios::end); 
+        writer << "end of the file";
+        writer.flush();
+        writer.close();
+
+        illegal_build_config = build_conf; // init
+        illegal_build_config.data_path = test_data_path;
+        illegal_cfg.clear();
+        knowhere::DiskANNBuildConfig::Set(illegal_cfg, illegal_build_config);
+        knowhere::IndexDiskANN<float> test_diskann(test_index_dir, metric_,
+                                                   std::make_unique<knowhere::LocalFileManager>());
+        EXPECT_THROW(test_diskann.BuildAll(nullptr, illegal_cfg), knowhere::KnowhereException);
+    }
+    fs::remove_all(test_dir);
```

Comment:
remove all will only remove data in this folder, you need to call remove after this to remove the folder.